app/src/main/java/au/com/bluedot/minimalintegration/MainApplication.java

```diff
@@ -85,138 +77,53 @@ private void requestPermissions() {
+                        return;
+                    }
+                    Toast.makeText(getApplicationContext(),"GeoTrigger started successfully",Toast.LENGTH_LONG).show();
+
+                });
     }
 
-    /**
-     * This callback happens when user is subscribed to Application Notification
-     * and check into any beacon under that Zone
-     *
-     * @param beaconInfo - Beacon triggered
-     * @param zoneInfo   - Zone information Beacon belongs to
-     * @param location   - geographical coordinate where trigger happened
-     * @param proximity  - the proximity at which the trigger occurred
-     * @param customData - custom data associated with this Custom Action
-     * @param isCheckOut - CheckOut will be tracked and delivered once device left the Beacon advertisement range
-     */
-    @Override
-    public void onCheckIntoBeacon(final BeaconInfo beaconInfo, ZoneInfo zoneInfo, LocationInfo location, Proximity proximity, Map<String, String> customData, boolean isCheckOut) {
-        //Using handler to pass Runnable into UI thread to interact with UI Elements
-        handler.post(new Runnable() {
-            @Override
-            public void run() {
-                Toast.makeText(getApplicationContext(), "Entered: " + beaconInfo.getName(), Toast.LENGTH_LONG).show();
+    void stopGeoTrigger() {
+        GeoTriggeringService.stop(getApplicationContext(), bdError -> {
+            String text = "GeoTrigger stop ";
+            if (bdError != null) {
+                text = text + bdError.getReason();
+            } else {
+                text = text + "Success ";
             }
+            Toast.makeText(getApplicationContext(), text, Toast.LENGTH_LONG).show();
         });
     }
 
-    /**
-     * This callback happens when user is subscribed to Application Notification
-     * and checked out from beacon under that Zone
-     *
-     * @param beaconInfo - Beacon is checked out from
-     * @param zoneInfo   - Zone information Beacon belongs to
-     * @param dwellTime  - time spent inside the Beacon area; in minutes
-     * @param customData - custom data associated with this Custom Action
-     */
-    @Override
-    public void onCheckedOutFromBeacon(final BeaconInfo beaconInfo, ZoneInfo zoneInfo, final int dwellTime, Map<String, String> customData) {
-        //Using handler to pass Runnable into UI thread to interact with UI Elements
-        handler.post(new Runnable() {
-            @Override
-            public void run() {
-                Toast.makeText(getApplicationContext(), "Left: " + beaconInfo.getName() + " dwellTime,min=" + dwellTime, Toast.LENGTH_LONG)
-                        .show();
-            }
-        });
+    void startTempo(){
+        TempoService.builder()
+                .notification(createNotification())
+                .destinationId("<DESTINATION_ID>")
```

Comment:
Let's also make the tempo destinationId a constant at the top of the file with a constant